parachains/runtimes/assets/statemint/src/constants.rs

```diff
@@ -55,13 +56,45 @@ pub mod fee {
 	///   - Setting it to `0` will essentially disable the weight fee.
 	///   - Setting it to `1` will cause the literal `#[weight = x]` values to be charged.
 	pub struct WeightToFee;
-	impl WeightToFeePolynomial for WeightToFee {
+	impl frame_support::weights::WeightToFee for WeightToFee {
```

Comment:
Could be simplified by implementing `WeightToFee ` for `Tuple`? or too many tuples impls? 🙈

Reply:
For now it is fine or? Since this targets 0.9.38.1

Reply:
Yeah nah, you would be confused if it is addition, or max or what. 

I suppose we can provide some nice helpers at he substrate layer though: 

```
struct RefTimeWeightCostPerMicroSec<Balance, Cost: Get<Balance>>(..);
impl WeightToFee for RefTimeConstPerMicroSec { .. }

struct ProofSizeWeightCostPerKByte<Balance, Cost: Get<Balance>>(..);
impl WeightToFee for RefTimeConstPerMicroSec { .. }
```

Or even a more generic: 

```
struct CostPerUnit<Balance, Cost: Get<Balance>, Unit: ops::Div<Balance>>; 
```

I think you can see where I am going with this. Just an idea

Reply:
> For now it is fine or? Since this targets 0.9.38.1

Yeah fine for now. Perhaps a follow-up issue that you can mentor?